### ISSUE TITLE
Fails to respawn when launched via Visual Studio Code debug

### ISSUE THREAD

**Issue by kenlyon** [whitecolor/ts-node-dev#64]
This may be a bug in [Visual Studio Code](https://code.visualstudio.com/), but I thought I'd post here first in case I'm just missing something with the configuration.

## What DOES work: run via terminal

My `package.json` contains this entry in the `scripts` element:
```
"debug": "ts-node-dev --inspect --respawn --transpileOnly ./src/server.ts"
```
I run it from the terminal like so:
```
npm run debug
```
This works well - I can add breakpoints and debug along the way. Whenever I edit and save a file, it respawns the process and the debugger re-attaches. All good!

The only limitation with this approach is that I need to go to the terminal and type the command each time. Since I develop in _Visual Studio Code_, I would like to use its debug configuration, which can be easily invoked by pressing `F5`. This leads to what doesn't work...

## What doesn't work: Debug via _Visual Studio Code_

_Visual Studio Code_ supports the use of [launch configurations](https://code.visualstudio.com/docs/editor/debugging#_launch-configurations). My `launch.json` contains this configuration entry:

```
{
    "type": "node",
    "request": "launch",
    "name": "Launch via NPM",
    "runtimeExecutable": "npm",
    "runtimeArgs": [
        "run-script",
        "debug"
    ],
    "port": 9229
}
```

When I use this approach, things start out ok - the debugger starts the process and I can add breakpoints and debug the code. However, whenever I edit a file and save it, I see the "Restarting" toast window appear, but the process fails to restart

NOTE: The same happens if I configure `launch.json` to invoke the `ts-node-dev` command directly. I show the above example here as it makes it clearer we're dealing with exactly the same parameters.

## The Question

Is there something more that `ts-node-dev` should be doing to place nicely with _Visual Studio Code_, or do you think the problem lies in their application?

**Comment by whitecolor** [comment]
No sure, don't work with such config.

**Comment by kenlyon** [comment]
@whitecolor Is this a setup that you expect to support, though? I think a lot of people use Visual Studio Code for their Typescript development.

**Comment by whitecolor** [comment]
There were some closed issues about using ts-node-dev with VSCode debugger, look at them, please.

**Comment by kenlyon** [comment]
The only issue I saw that was similar was #9. I tried the proposed solutions there, but none of them resolved my problem.

**Comment by Sharpiro** [comment]
I agree it would be nice to debug from vs code by just using a launch configuration. 

Right now the launch configuration crashes on code change. 

So right now you must run debug command and then attach vs code to the port

**Comment by vegerot** [comment]
@Sharpiro what's an example config to do that?

**Comment by Sharpiro** [comment]
Created a working example of this bug and how starting `ts-node-dev` from shell first works, but if vs-code launches it instead of just attaching, it will crash on code change.

https://github.com/Sharpiro/bugs/tree/master/ts-node-dev/vs-code-debugging

specifically the launch file
https://github.com/Sharpiro/bugs/blob/master/ts-node-dev/vs-code-debugging/.vscode/launch.json

**Comment by mpcen** [comment]
Any plans to fix this?
